crates/oxc_linter/src/rules/eslint/no_new_func.rs

```diff
@@ -0,0 +1,139 @@
+use oxc_ast::{ast::IdentifierReference, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_new_func(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("The Function constructor is eval.").with_label(span)
+}
+
+fn is_global_function_reference(ctx: &LintContext, id: &IdentifierReference) -> bool {
+    if let Some(reference_id) = id.reference_id() {
+        return id.name == "Function" && ctx.symbols().is_global_reference(reference_id);
+    }
+    false
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNewFunc;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// The rule disallow `new` operators with the `Function` object.
```

Comment:
Please fill in the "why this is bad" section

Reply:
Thank you for your guidance. This is a bad habit left by previous rule writing, and I will complete all the missing rules.